File: src/types/media.ts
export type AssetStatus = 'fetching' | 'ready' | 'error';

export type AssetType = 'image' | 'audio';

export type Asset = {
  id: string;
  type: AssetType;
  name: string;
  sourceUrl?: string;       // original remote URL if imported from URL
  localPath?: string;       // absolute path to cached/local file
  contentHash?: string;     // sha256 of the local file
  fetchedAt?: string;       // ISO 8601 timestamp of last successful download
  status: AssetStatus;
  error?: string;
  // media metadata, populated after ffprobe
  duration?: number;        // seconds (audio / video)
  width?: number;           // pixels (image)
  height?: number;          // pixels (image)
  sampleRate?: number;      // hz (audio)
  channels?: number;        // channel count (audio)
};

File: src-tauri/src/models/media.rs
use serde::{Deserialize, Serialize};

/// Returned by `probe_media` — ffprobe-derived media metadata.
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MediaInfo {
    pub kind: String,              // "image" | "audio"
    pub duration: Option<f64>,
    pub width: Option<u32>,
    pub height: Option<u32>,
    pub sample_rate: Option<u32>,
    pub channels: Option<u32>,
}

/// Returned by `download_asset` after a successful remote fetch.
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DownloadResult {
    pub local_path: String,
    pub content_hash: String,
    pub fetched_at: String,   // ISO 8601
}

/// Version info returned by capability checks.
#[derive(Debug, Serialize, Deserialize)]
pub struct ToolInfo {
    pub available: bool,
    pub version: String,
}
